Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/RdbmsProviderTest.cs

```diff
@@ -67,11 +67,55 @@ public override void SetUp ()
 
       _provider = new TestableRdbmsProvider(
           TestDomainStorageProviderDefinition,
+          TestDomainStorageProviderDefinition.ConnectionString,
           NullPersistenceExtension.Instance,
           _commandFactoryMock.Object,
           () => _connectionCreatorMock.Object.CreateConnection());
     }
 
+    [Test]
+    public void RdbmsProvider_WithConnectionStringContainedInProviderDefinition ()
+    {
+      var rdbmsProvider = new RdbmsProvider(
+          new RdbmsProviderDefinition("dummyProviderDefinition", Mock.Of<IRdbmsStorageObjectFactory>(), "dummyConnectionString", "dummyReadOnlyConnectionString"),
+          "dummyConnectionString",
+          NullPersistenceExtension.Instance,
+          _commandFactoryMock.Object,
+          () => _connectionCreatorMock.Object.CreateConnection());
+
+      Assert.That(rdbmsProvider.ConnectionString, Is.EqualTo("dummyConnectionString"));
+    }
+
+    [Test]
+    public void RdbmsProvider_WithReadOnlyConnectionStringContainedInProviderDefinition ()
+    {
+      var rdbmsProvider = new RdbmsProvider(
+          new RdbmsProviderDefinition("dummyProviderDefinition", Mock.Of<IRdbmsStorageObjectFactory>(), "dummyConnectionString", "dummyReadOnlyConnectionString"),
+          "dummyReadOnlyConnectionString",
+          NullPersistenceExtension.Instance,
+          _commandFactoryMock.Object,
+          () => _connectionCreatorMock.Object.CreateConnection());
+
+      Assert.That(rdbmsProvider.ConnectionString, Is.EqualTo("dummyReadOnlyConnectionString"));
+    }
+
+    [Test]
+    public void RdbmsProvider_WithConnectionStringNotInProviderDefinition_ThrowsArgumentException ()
```

Comment:
don't we need a test for readonly, too?

Reply:
No, because the connection string must be either the read-write oder the read-only connection string in the `RdbmsProviderDefinition`. This test case covers if it is neither.

Reply:
ah, it's *that* test. my bad.